partiql-logical-planner/src/lower.rs

```diff
@@ -582,21 +603,30 @@ impl<'a, 'ast> Visitor<'ast> for AstToLogical<'a> {
 
     fn enter_top_level_query(&mut self, _query: &'ast ast::TopLevelQuery) -> Traverse {
         self.enter_benv();
+        self.plan_stack.push(LogicalPlan::default());
         Traverse::Continue
     }
     fn exit_top_level_query(&mut self, _query: &'ast ast::TopLevelQuery) -> Traverse {
         let mut benv = self.exit_benv();
         eq_or_fault!(self, benv.len(), 1, "Expect benv.len() == 1");
         let out = benv.pop().unwrap();
-        let sink_id = self.plan.add_operator(BindingsOp::Sink);
-        self.plan.add_flow(out, sink_id);
+        let plan = cur_plan_or_fault!(self);
+        let sink_id = plan.add_operator(BindingsOp::Sink);
+        plan.add_flow(out, sink_id);
         Traverse::Continue
     }
 
     fn enter_query(&mut self, query: &'ast Query) -> Traverse {
         self.enter_benv();
         if let QuerySet::Select(_) = query.set.node {
             self.enter_q();
+            if self.q_stack.len() > 1 {
+                // entering subquery
+                // TODO: currently there's no enforcement for unique `OpId` between the outer query
+                //  plan and the subquery plan. To decide if we want the `OpId`s to be unique
+                //  between any subqueries.
```

Comment:
The subquery plan and outer query plan don't currently enforce unique `OpId`s between the two plans. I was going off of how subqueries were modeled in the logical and eval plan as part of #227 and [example tests](https://github.com/partiql/partiql-lang-rust/blob/9366938e3c63b01a1035d10dbb6a4905d45ffbd7/partiql-eval/src/lib.rs#L2042-L2175).

I was wondering if we want unique `OpId`s between the two plans, which can be helpful for debugging and plan graph visualization.